### ISSUE TITLE
Clarification on Input Dimensions and Feature-based Iteration in model

### ISSUE THREAD

**Issue by S1mple0107** [muslehal/xLSTMTime#19]
Thanks for your project and time!!
i find that the input in the project is (batch_size, features, context_points). for example, etth1 dataset's input is (batch, features=7, context_points=512).
sLSTMBlock's input is (batch, features=7, embedding_dim=256), which cause to the steps in slstm_forward.cu is featrues=7. According to the principle of RNN-type models, this iteration count should be related to the time steps.
Through my experiments, I found that your method can achieve good results. I don't know why the loop operation based on variables can yield good outcomes. I think the input should be (batch, context_points=512, features=7). By using slstm, the latest output h_t hidden state is obtained, and then the target output (batch, target_points, features) is obtained through multi-task learning.

I noticed that the loss value of the validation set reached its optimum in the first few epochs and then rose monotonically thereafter. Considering this is an indication of overfitting, do you have a similar situation?

could you give me some advices?

**Comment by Kryptodefii** [comment]
@S1mple0107 hey mate, have you been able to open ticket number with the support?

**Comment by S1mple0107** [comment]
> [@S1mple0107](https://github.com/S1mple0107) hey mate, have you been able to open ticket number with the support?

yes, i run this project successfully

**Comment by xinyanX** [comment]
I have the same question as @S1mple0107. Before reading the code, I thought straightforward it should be in (batch, seq_len, n_dims) as common for RNNs. In this case (batch, features, context_points) for xLSTMTime, it is in this aspact the same as the settings for another xLSTM-based work xLSTM-Mixer ([https://arxiv.org/html/2410.16928v1]). I wonder if it is better empirically for xLSTM to just stride over features rather than time, on top of the time-mixing modules before the xLSTM blocks?

**Comment by S1mple0107** [comment]
> I have the same question as [@S1mple0107](https://github.com/S1mple0107). Before reading the code, I thought straightforward it should be in (batch, seq_len, n_dims) as common for RNNs. In this case (batch, features, context_points) for xLSTMTime, it is in this aspact the same as the settings for another xLSTM-based work xLSTM-Mixer ([https://arxiv.org/html/2410.16928v1]). I wonder if it is better empirically for xLSTM to just stride over features rather than time, on top of the time-mixing modules before the xLSTM blocks?

**Comment by S1mple0107** [comment]
> I have the same question as [@S1mple0107](https://github.com/S1mple0107). Before reading the code, I thought straightforward it should be in (batch, seq_len, n_dims) as common for RNNs. In this case (batch, features, context_points) for xLSTMTime, it is in this aspact the same as the settings for another xLSTM-based work xLSTM-Mixer ([https://arxiv.org/html/2410.16928v1]). I wonder if it is better empirically for xLSTM to just stride over features rather than time, on top of the time-mixing modules before the xLSTM blocks?

For SLSTM, its improvement compared to the LSTM model lies only in the new information mixing mechanism, activation function, and stable normalization. The principle remains unchanged: the RNN model updates the hidden state iteratively based on time steps. Therefore, I think the feature dimension iteration is unreasonable and I don't know why it has such an effect.
